doctr/utils/geometry.py

```diff
@@ -145,54 +145,101 @@ def rotate_abs_boxes(boxes: np.ndarray, angle: float, img_shape: Tuple[int, int]
+
+
 def rotate_boxes(
     boxes: np.ndarray,
     angle: float = 0.,
-    min_angle: float = 1.
+    min_angle: float = 1.,
+    orig_shape: Optional[Tuple[int, int]] = None,
+    mask_shape: Optional[Tuple[int, int]] = None,
 ) -> np.ndarray:
-    """Rotate a batch of straight bounding boxes (xmin, ymin, xmax, ymax) of an angle,
-    if angle > min_angle, around the center of the page.
+    """Rotate a batch of straight bounding boxes (xmin, ymin, xmax, ymax) or rotated bounding boxes
+    (x, y, w, h, alpha) of an angle, if angle > min_angle, around the center of the page.
 
     Args:
-        boxes: (N, 4) array of RELATIVE boxes
+        boxes: (N, 4) or (N, 5) array of RELATIVE boxes
         angle: angle between -90 and +90 degrees
         min_angle: minimum angle to rotate boxes
+        orig_shape: shape of the origin image
+        mask_shape: shape of the mask if the image is cropped after the rotation
 
     Returns:
         A batch of rotated boxes (N, 5): (x, y, w, h, alpha) or a batch of straight bounding boxes
     """
+    # Change format of the boxes to rotated boxes
+    _boxes = boxes.copy()
+    if _boxes.shape[1] == 4:
+        _boxes = np.column_stack(((_boxes[:, 0] + _boxes[:, 2]) / 2,
+                                  (_boxes[:, 1] + _boxes[:, 3]) / 2,
+                                  _boxes[:, 2] - _boxes[:, 0],
+                                  _boxes[:, 3] - _boxes[:, 1],
+                                  np.zeros(_boxes.shape[0])))
     # If small angle, return boxes (no rotation)
     if abs(angle) < min_angle or abs(angle) > 90 - min_angle:
-        return boxes
+        return _boxes
     # Compute rotation matrix
     angle_rad = angle * np.pi / 180.  # compute radian angle for np functions
     rotation_mat = np.array([
         [np.cos(angle_rad), -np.sin(angle_rad)],
         [np.sin(angle_rad), np.cos(angle_rad)]
-    ], dtype=boxes.dtype)
-    # Compute unrotated boxes
-    x_unrotated, y_unrotated = (boxes[:, 0] + boxes[:, 2]) / 2, (boxes[:, 1] + boxes[:, 3]) / 2
-    width, height = boxes[:, 2] - boxes[:, 0], boxes[:, 3] - boxes[:, 1]
+    ], dtype=_boxes.dtype)
     # Rotate centers
-    centers = np.stack((x_unrotated, y_unrotated), axis=-1)
-    rotated_centers = .5 + np.matmul(centers - .5, np.transpose(rotation_mat))
+    centers = np.stack((_boxes[:, 0], _boxes[:, 1]), axis=-1)
+    rotated_centers = .5 + np.matmul(centers - .5, rotation_mat)
     x_center, y_center = rotated_centers[:, 0], rotated_centers[:, 1]
     # Compute rotated boxes
-    rotated_boxes = np.stack((x_center, y_center, width, height, angle * np.ones_like(boxes[:, 0])), axis=1)
+    rotated_boxes = np.stack((x_center, y_center, _boxes[:, 2], _boxes[:, 3], angle * np.ones_like(_boxes[:, 0])),
+                             axis=1)
+    # Apply a mask if requested
+    if mask_shape is not None and orig_shape is not None:
+        rotated_boxes = remap_boxes(rotated_boxes, orig_shape=orig_shape, dest_shape=mask_shape)
```

Comment:
I think we need to explain somewhere (perhaps in arg description) what happens when either of those is None

Reply:
OK I added some description that was indeed missing